### DISCUSSION TITLE
Playing music via streaming from the URL in an API using React Native and Expo

### DISCUSSION THREAD

**Discussion by eerjuano** [expo/expo#23935] [Archived: Questions]
I'm opening this discussion in case someone can help me with this little issue I'm facing. Expo Go has its audio playback library, audio-av, but the problem is that it only plays direct links to .mp3 files. An official API doesn't provide access to the file's URI, but rather an online streaming link. In my search, I found that react-native-track-player does offer that possibility, but unfortunately, it's not compatible with Expo Go. Do you know of any way or library that is compatible with Expo Go and provides the option to play API links? I'm looking forward to your input.

This is what I get when I try with react-native-track-player or react-native-sound:

ERROR TypeError: Cannot read property 'IsAndroid' of null, js engine: hermes
ERROR Invariant Violation: "main" has not been registered. This can happen if:

Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
A module failed to load due to an error and AppRegistry.registerComponent wasn't called., js engine: hermes

**Comment by tsapeta** [comment]
I think in this case Expo Go is just not enough and you should level up by using development builds to have the native code of custom libraries, that are not included in Expo Go.
You can read more about development builds here: https://docs.expo.dev/develop/development-builds/introduction/

Besides that, thanks for your feedback that this feature is missing in expo-av 🙏 We're planning to split it into expo-audio and expo-video, re-implement them from scratch. I'll make sure to include this feature then!

**Reply by eerjuano** [reply]
Thanks for your answer, I've been researching for the resolution of my problem or a clue and I read something of development builds. I appreciate to much the feedbacks of expo.

**Reply by jwhijazi** [reply]
@tsapeta @eerjuano 
I'm trying to build a radio player app using expo-av (url streaming), I have the same problem; createAsync takes long time to load the stream and start playing it, unlike mp3 files which plays directly.
Hope this will be solved in the next version of expo-av.

Thanks.

**Reply by eerjuano** [reply]
Hi @jwhijazi, I left that project that I was creating because the main problem doesn`t was the expo-av. The main problem was that I tried to reproduce links with the ID of the music track of deezer, and not the url streaming. Besides, I don't use expo since 4 months ago so I'm not the best option to help you. Make sure the problem it is not from the api that gives the url streaming and have bigs latencies. 

All the best,
eerjuano.

**Reply by jwhijazi** [reply]
@eerjuano 
Thanks for the reply, I'm waiting for expo-av to be updated and seperate audio from video packages, as we hope streaming issues will be fixed.

**Reply by balanmihai** [reply]
@jwhijazi im facing the same problem. i want to build aswell a radio application that gets the audio from an url. did you manage to solve this issue? please contact me if you can

**Reply by SimonTart** [reply]
@tsapeta 
I found that the Expo Audio library has already been released, but I tried it and couldn't play streaming music. Does Expo Audio currently support streaming music playback?

**Reply by AppleJem** [reply]
No it does not, unfortunately that's just a use case that isn't supported by expo Audio yet

> @tsapeta I found that the Expo Audio library has already been released, but I tried it and couldn't play streaming music. Does Expo Audio currently support streaming music playback?
